blocksuite/affine/gfx/turbo-renderer/src/variants/paragraph/paragraph-layout-provider.ts

```diff
@@ -0,0 +1,89 @@
+import type { GfxBlockComponent } from '@blocksuite/block-std';
+import { clientToModelCoord } from '@blocksuite/block-std/gfx';
+
+import { BlockLayoutProvider } from '../../layout/block-layout-provider';
+import type { Rect } from '../../types';
+import type { ParagraphLayout } from './paragraph-painter';
+import { getSentenceRects, segmentSentences } from './text-utils';
+
+export class ParagraphLayoutProvider extends BlockLayoutProvider<ParagraphLayout> {
```

Comment:
Can this be put in the paragraph block package?

Reply:
Will do in next PR. The `variants` is somewhat needed for sharing deps between paragraph / list / code blocks, etc.